### ISSUE TITLE
DER encoded `BIT STRING` produces `BitString` with incorrect length

### ISSUE THREAD

**Issue by benmaddison** [XAMPPRocky/rasn#134]
Decoding a DER-encoded `BIT STRING` with a non-zero "unused bits" byte into a `types::BitString` results in a value padded with `0`s to the nearest byte.

This makes it impossible to tell the difference between values with "real" trailing zeros and those without.

In particular, [RFC3779] and other RPKI-related standards use a `BIT STRING` to encode IP prefixes, indicating the length of the prefix through the bit-length of the `BIT STRING`. This issue makes it impossible to correctly decode IP prefixes with lengths that are not a multiple of 8.

Note that I have *only* tried this with the DER implementation. I do not know whether the other codecs suffer from a similar issue.

The issue can be reproduced with the following minimal example:

``` rust
use bitvec::{bitvec, order::Msb0};
use rasn::{der, types::BitString};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // DER-encoded byte-slice of BIT STRING B'0101'
    let encoded: &[u8] = &[
        0x03,        // tag = 3
        0x02,        // length = 2
        0x04,        // unused bits = 4
        0b0101_0000, // value
    ];
    let bit_string = der::decode::<BitString>(encoded)?;

    // Expected value
    let expected = bitvec![u8, Msb0; 0, 1, 0, 1];

    assert_eq!(bit_string, expected);

    Ok(())
}
```

I will take a look at the relevant code, and propose a fix if I can grok it easily enough.

[RFC3779]: https://datatracker.ietf.org/doc/html/rfc3779#section-2.2.3.8

**Comment by XAMPPRocky** [comment]
Thank you for your issue! Just to check. Have you tested with the latest version? I'm pretty sure I fixed this last week https://github.com/XAMPPRocky/rasn/commit/39a9f4758c060997028e76fb59e10fa187af0660

**Comment by XAMPPRocky** [comment]
Ah nvm, it seems I didn't add back in the intentional bit string truncation. So it should be simple to just adjust the decode and encode to do that.

> Note that I have only tried this with the DER implementation. I do not know whether the other codecs suffer from a similar issue.

This is only relevant for BER, PER bit strings don't have the concept of extra bits since it is a bit-level codec.

**Comment by benmaddison** [comment]
@XAMPPRocky have attempted a fix in #135.
Let me know your thoughts when you have a sec.
I have tested on my project, and this fixes the bug that lead me here.
